pkgs/tools/misc/trac/default.nix

```
@@ -0,0 +1,52 @@
+{ lib
+, buildPythonApplication
+, fetchPypi
+, Babel
+, docutils
+, pygments
+, pytz
+, jinja2
+, pysqlite
+, psycopg2
+, pymysql
+, git
+, setuptools
+}:
+
+
+buildPythonApplication rec {
+  pname = "trac";
+  version = "1.4";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "Trac";
+    sha256 = "1cg51rg0vb9vf23wgn28z3szlxhwnxprj5m0mvibqyypi123bvx1";
+  };
+
+  prePatch = ''
+    # Removing the date format tests as they are outdated
+    substituteInPlace trac/util/tests/__init__.py \
+      --replace "suite.addTest(datefmt.test_suite())" ""
+  '';
+
+  propagatedBuildInputs = [
+    Babel
+    docutils
+    pygments
+    pytz
+    jinja2
+    pysqlite
+    psycopg2
+    pymysql
+    git
+    setuptools
+  ];
+
+  meta = with lib; {
+    description = "Integrated SCM, wiki, issue tracker and project environment";
+    homepage = "https://trac.edgewall.org/";
+    license = licenses.bsd2;
```

```suggestion
    license = licenses.bsd3;
```

https://trac.edgewall.org/browser/tags/trac-1.4/COPYING

---

I think this is ```BSD License 2```, the third one mentions advertising materials or am I wrong?

https://en.wikipedia.org/wiki/BSD_licenses#3-clause_license_(%22BSD_License_2.0%22,_%22Revised_BSD_License%22,_%22New_BSD_License%22,_or_%22Modified_BSD_License%22)

IANAL, please advise if you can.

---

Indeed, it's a bit confusing, BSD License 2 = 3-Clause BSD License = [`licenses.bsd3`](https://github.com/NixOS/nixpkgs/blob/47bd85469fbd507c12d8ec7f9650ab6dcbd8c274/lib/licenses.nix#L98) in nixpkgs.

`licenses.bsd2` is the 2-Clause BSD License, which is the next license down on your wikipedia link. Or in other words, `licenses.bsd2` != BSD License 2, which is the source of confusion. :sweat_smile:

---

Ooooh, got it - thank you! :)